### PR TITLE
Add form validation and feedback

### LINKED ISSUE #97: As a organization member, I want to know if my form submission was successful, so that I know if I have to try again
- [ ] Add feedback to the add game form
- [ ] Add feedback to the add platform form
- [ ] Add feedback to the borrow request form
- [ ] Add feedback to the suggestions form

### LINKED ISSUE #99: As a dumb user, I want to get better error responses to the forms, so that I know what i did wrong
Update All router's responses so that the error codes and messages reflect their use case

### PR DESCRIPTION
**Pull request by Oscariremma**
## Key features
- Adds zod form validation to the user facing forms
- Adds toasts that indicate success or fail of a submission
- Moves more API requests over to React Querry
- Closes #97 
- Closes #99 

## Known issues
- The new inputs need proper CSS styling

### CODE REVIEW

File: frontend/src/components/AddGame/AddGame.tsx [hunk 1]
```diff
@@ -38,93 +85,124 @@ const AddGame: FC<AddGameProps> = () => {
...
+								}
+							);
+							resetForm();
+						} catch (e) {}
+					},
+					(e) =>
+						toast.error('Please fill out all fields correctly!', {
+							position: 'bottom-right'
+						})
+				)}
```
Comment (Supergamer1337) [inline frontend/src/components/AddGame/AddGame.tsx:121]: Please move this to separate function, not inline

File: frontend/src/components/AddPlatform/addPlatform.tsx [hunk 2]
```diff
@@ -1,36 +1,85 @@
...
+							}
+						);
+						resetForm();
+					} catch (e) {}
+				},
+				(e) =>
+					toast.error('Please fill out all fields correctly!', {
+						position: 'bottom-right'
+					})
+			)}
```
Comment (Supergamer1337) [inline frontend/src/components/AddPlatform/addPlatform.tsx:71]: Move to separate function

File: frontend/src/components/AddSuggestion/AddSuggestion.tsx [hunk 3]
```diff
@@ -35,85 +83,122 @@ const AddSuggestion: FC<AddSuggestionProps> = () => {
...
+							}
+						);
+						resetForm();
+					} catch (e) {}
+				},
+				(e) =>
+					toast.error('Please fill out all fields correctly!', {
+						position: 'bottom-right'
+					})
+			)}
```
Comment (Supergamer1337) [inline frontend/src/components/AddSuggestion/AddSuggestion.tsx:117]: At this point you know

**Review by Supergamer1337 — CHANGES_REQUESTED** [review]
Some code could be made more readable, otherwise good

File: frontend/src/components/AddGame/AddGame.tsx (hunk 1 above)
Comment (Oscariremma) [reply frontend/src/components/AddGame/AddGame.tsx]: Better now?

**Comment by Oscariremma** [comment]
> The code itself looks good, but I'd like to get toasts when creating borrow requests and approving them too. image

This was quite a lot of work as basically everything I needed to change required quite some refactoring....
1000+ lines changed go brrrrrr!
image

**Comment by x183** [comment]
> > The code itself looks good, but I'd like to get toasts when creating borrow requests and approving them too. image
> 
> This was quite a lot of work as basically everything I needed to change required quite some refactoring....
> 1000+ lines changed go brrrrrr!
> image
> 

Sorry about that! 😅

**Review by Supergamer1337 — APPROVED** [review]
Lgtm

**Review by x183 — APPROVED** [review]
Nice work

**Pull request merged**